src/lando/main/scm/git.py

```diff
@@ -0,0 +1,258 @@
+            f"{ENV_COMMITTER_EMAIL} and {ENV_COMMITTER_NAME} set to {requester_email}"
+        )
+        try:
+            yield self
+        finally:
+            del os.environ[ENV_COMMITTER_NAME]
+            del os.environ[ENV_COMMITTER_EMAIL]
+
+    def head_ref(self) -> str:
+        """Get the current revision_id"""
+        return self._git_run("rev-parse", "HEAD", cwd=self.path)
+
+    def changeset_descriptions(self) -> list[str]:
+        """Retrieve the descriptions of commits in the repository.
+
+        Returns:
+            list[str]: A list of first lines of changeset descriptions.
+        """
+        command = ["log", "--format=%s", "@{u}.."]
+        return self._git_run(*command).splitlines()
+
+    def update_repo(self, pull_path: str, target_cset: Optional[str] = None) -> str:
+        """Update the repository to the specified changeset.
+
+        This method uses the Git commands to update the repository
+        located at the given pull path to the specified target changeset.
+
+        Args:
+            pull_path (str): The path to pull from.
+            target_cset (str): The target changeset to update the repository to.
+
+        Returns:
+            str: The target changeset
+        """
+        branch = target_cset or self.default_branch
+        self.clean_repo()
+        self._git_run("pull", "--prune", pull_path)
+        self._git_run("checkout", "--force", "-B", branch)
+        return self.head_ref()
+
+    def clean_repo(self, *, strip_non_public_commits: bool = True):
+        """Reset the local repository to the origin"""
+        if strip_non_public_commits:
+            self._git_run(
+                "reset", "--hard", f"origin/{self.default_branch}", cwd=self.path
+            )
+        self._git_run("clean", "-fdx", cwd=self.path)
+
+    def format_stack_amend(self) -> Optional[list[str]]:
+        """Amend the top commit in the patch stack with changes from formatting."""
+        self._git_run("commit", "--all", "--amend", "--no-edit")
+        return [self.get_current_node()]
+
+    def format_stack_tip(self, commit_message: str) -> Optional[list[str]]:
+        """Add an autoformat commit to the top of the patch stack.
+
+        Return the commit hash of the autoformat commit as a `str`,
+        or return `None` if autoformatting made no changes.
+        """
+        self._git_run("commit", "--all", "--message", commit_message)
+        return [self.get_current_node()]
+
+    def get_current_node(self) -> str:
+        """Return the commit_id of the tip of the current branch"""
+        return self._git_run("rev-parse", "HEAD")
+
+    @property
+    def repo_is_initialized(self) -> bool:
+        """Determine whether the target repository is initialised"""
+        if not Path(self.path).exists():
+            return False
+
+        try:
+            self._git_run("status", cwd=self.path)
+        except SCMException:
+            return False
+
+        return True
```

Comment:
I think a more positive check here would be good too, for example, checking the status code of the `git` command.

Reply:
`_git_run` will throw an SCMException on any non-0 status.

https://github.com/mozilla-conduit/lando/pull/147/files/635c25d299264d79adb887f6794b8e21c1d3f212#diff-2337f63ffc234b7c7cbd13e6ab6f4e1122832768b9ea5250508f5830a5bf71e5R247-R251